### ISSUE TITLE
silent crash to desktop due to unhandled exception

### ISSUE THREAD

**Issue by ghramsey** [MattFiler/OpenCAGE#303]
While editing got a silent crash to desktop.

The windows event viewer reported .NET was closed and fault was an unhandled exception.
I am not sure if the event log information will help, but for what it's worth.

Of note here is that I'm been noticing lots of delay in the tabbed window.  
It will draw slow enough I get window tearing (only on the tabs) while it's drawing.

I mention this as prior to the crash the window was in process of drawing slowly then it closed the editor.
I know you said that delays while loading many tabs was a known issue, so I've refrained from making an unnecessary ticket.  
However, it does seem like the delay issue gets worse and in some cases it gets the commands editor to 'not responding' while 
it's redrawing the window.   This is the one and only crash I've seen though.    I have also taken to using the close all tabs option routinely, but this does not seem to have any affect on the slow drawng.

I have option settings>
Use file browser viewer off, 
Auto hide composite viewer is off, 
Open entities in new tabs is also off(by default? I don't remember changing this), 
Keep function window open is checked.

My system consists of an I7-9700 16GB RAM and an RTX3060. RTX drivers just updated on Wed.  A quad of 1TB SSDs.
I usually cap to 200 FPS in the driver options since it runs cooler that way, but I make no other edits on the configuration.
I'm still on Windows 10 Home edition.

```
=
[-](https://github.com/MattFiler/OpenCAGE/issues/new#) <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
[-](https://github.com/MattFiler/OpenCAGE/issues/new#) <System>
  <Provider Name="Application Error" /> 
  <EventID Qualifiers="0">1000</EventID> 
  <Version>0</Version> 
  <Level>2</Level> 
  <Task>100</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2023-10-12T23:57:47.5873246Z" /> 
  <EventRecordID>167962</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="0" ThreadID="0" /> 
  <Channel>Application</Channel> 
  <Computer>ghram</Computer> 
  <Security /> 
  </System>
[-](https://github.com/MattFiler/OpenCAGE/issues/new#) <EventData>
  <Data>CommandsEditor.exe</Data> 
  <Data>0.2.0.0</Data> 
  <Data>99c21269</Data> 
  <Data>KERNELBASE.dll</Data> 
  <Data>10.0.19041.3516</Data> 
  <Data>e9b06223</Data> 
  <Data>c000041d</Data> 
  <Data>000000000002cf19</Data> 
  <Data>3b8c</Data> 
  <Data>01d9fd24cc4c5941</Data> 
  <Data>D:\SteamLibrary\steamapps\common\Alien Isolation\DATA\MODTOOLS\REMOTE_ASSETS\scripteditor\CommandsEditor.exe</Data> 
  <Data>C:\WINDOWS\System32\KERNELBASE.dll</Data> 
  <Data>94150a53-c3a3-4b6d-98cf-2d7b96e69a79</Data> 
  <Data /> 
  <Data /> 
  </EventData>
  </Event>
=
[-](https://github.com/MattFiler/OpenCAGE/issues/new#) <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
[-](https://github.com/MattFiler/OpenCAGE/issues/new#) <System>
  <Provider Name="Application Error" /> 
  <EventID Qualifiers="0">1000</EventID> 
  <Version>0</Version> 
  <Level>2</Level> 
  <Task>100</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2023-10-12T23:57:44.1891136Z" /> 
  <EventRecordID>167960</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="0" ThreadID="0" /> 
  <Channel>Application</Channel> 
  <Computer>ghram</Computer> 
  <Security /> 
  </System>
[-](https://github.com/MattFiler/OpenCAGE/issues/new#) <EventData>
  <Data>CommandsEditor.exe</Data> 
  <Data>0.2.0.0</Data> 
  <Data>99c21269</Data> 
  <Data>System.Windows.Forms.ni.dll</Data> 
  <Data>4.8.9181.0</Data> 
  <Data>64b87db6</Data> 
  <Data>c0000005</Data> 
  <Data>0000000000a4d667</Data> 
  <Data>3b8c</Data> 
  <Data>01d9fd24cc4c5941</Data> 
  <Data>D:\SteamLibrary\steamapps\common\Alien Isolation\DATA\MODTOOLS\REMOTE_ASSETS\scripteditor\CommandsEditor.exe</Data> 
  <Data>C:\WINDOWS\assembly\NativeImages_v4.0.30319_64\System.Windows.Forms\f443d8df444525c5334b67cea18e81f6\System.Windows.Forms.ni.dll</Data> 
  <Data>75690952-ff96-4f8d-a735-f97ada8ef79b</Data> 
  <Data /> 
  <Data /> 
  </EventData>
  </Event>
=
[-](https://github.com/MattFiler/OpenCAGE/issues/new#) <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
[-](https://github.com/MattFiler/OpenCAGE/issues/new#) <System>
  <Provider Name=".NET Runtime" /> 
  <EventID Qualifiers="0">1026</EventID> 
  <Version>0</Version> 
  <Level>2</Level> 
  <Task>0</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2023-10-12T23:57:43.5841144Z" /> 
  <EventRecordID>167959</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="0" ThreadID="0" /> 
  <Channel>Application</Channel> 
  <Computer>ghram</Computer> 
  <Security /> 
  </System>
[-](https://github.com/MattFiler/OpenCAGE/issues/new#) <EventData>
  <Data>Application: CommandsEditor.exe Framework Version: v4.0.30319 Description: The process was terminated due to an unhandled exception. Exception Info: System.NullReferenceException</Data> 
  </EventData>
  </Event>
```

**Comment by MattFiler** [comment]
Hey, this is a bit of a vague report - were you doing anything specific when the crash happened?

**Comment by ghramsey** [comment]
> Hey, this is a bit of a vague report - were you doing anything specific when the crash happened?

Yeah. I'm sorry it's vague.  I was not doing anything, but normal editing. I've been going through animations 
and editing the playspeed parameters.  Nothing unusual.    Other than my notes about the tabs being slow  
to draw which had been happening while I was editing and was in progress of drawing them when it crashed 
I don't have more information.    

Do those event logs help at all?  I was not hopeful they would be, but since they existed I figured give them a try.

**Comment by MattFiler** [comment]
I'm going to consider this one fixed with all the recent performance improvements on staging, plus the latest bug fix for #330 which I think should solve this
